roles/pdf_gen/templates/opt/tempCleaner.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+
+""" 2018-02-26
+
+    tempCleaner.py
+
+    Author: Scott Batchelder
+
+    This tool is designed to clean out the /tmp/ directory on servers that
+    use that space, but do not clean up after themselves(pdf_gen). This
+    tool will only delete directories that have the pattern"/tmp/tmp*/"
+    and that are older than 7 days.  An entry for this tool should be added
+    to the 'crontab' so that it runs once per day.
+
+    Make sure to execute as the proper user:
+       % sudo -u www-data "./tempCleaner.py
+
+    version 1
+"""
+
+import os
+import time
+import shutil
+import re
+
+base = r"/tmp"
+now = time.time()
+
+for dirpath, dirnames, files in os.walk(base):
+    for dirname in dirnames:
+        if "tmp" in dirname.lower():
+            f = os.path.join(dirpath, dirname)
+            if os.stat(f).st_mtime < now - 7*86400:
+                shutil.rmtree(f)
+                # print "### Delete dir: " + f
+
+for g in os.listdir(base):
+    g = os.path.join(base, g)
+    r = re.compile('modexport|col')
+    s = re.compile('stderr|stdout')
+    if r.search(g):
+        if s.search(g):
+            if os.stat(g).st_mtime < now - 14*86400:
+                os.remove(g)
+                # print "### Delete: " + g
```

Comment:
This whole stanza feels not very DRY (Don't Repeat Yourself)
I'm thinking this should be structured as a function you call w/ 3 params: a base, a regex, and an age. You get back an iterator then operate in that. Config is at the top w/ a dictionary keyed on the regex, values are the age. __main__ function that loads config, loops on key,value pairs and deletes.

Reply:
On third read, I see that we're spec'ed a little tighter, this is handling two cases: tempdirs, and directly-in-base files. I guess this could stay as two stanzas, in that case. But do move the magic numbers into constants at the top